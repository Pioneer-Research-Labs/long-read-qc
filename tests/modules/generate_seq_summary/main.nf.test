nextflow_process {

    name "Test Process generate_seq_summary"
    script "modules/generate_seq_summary/main.nf"
    process "generate_seq_summary"

    test("Should generate a sequencing summary for two libraries") {

        when {
            params {
                outdir = outputDir
            }
            process {
                """
                input[0] = file('${projectDir}/test_data/seq_stats_map.tsv')
                input[1] = file('${projectDir}/test_data/barcode_map.tsv')
                input[2] = file('${projectDir}/test_data/mapped_vector_map.tsv')
                input[3] = file('${projectDir}/test_data/insert_map.tsv')
                """
            }
        }

        then {
            assert process.success
            assert snapshot(
            path("${params.outdir}/seq_summary.csv"),
            path("${params.outdir}/concatenated_seq_stats.csv"),
            path("${params.outdir}/concatenated_vector_map_stats.csv")).match()
        }

    }

}