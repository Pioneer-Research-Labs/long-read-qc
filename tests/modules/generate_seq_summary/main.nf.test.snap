{
    "Should generate a sequencing summary for two libraries": {
        "content": [
            "seq_summary.csv:md5,b37c915057aa36d19adfc8b1c1949650",
            "concatenated_seq_stats.csv:md5,d9f3f59a79b043974991ab81a5edaf50",
            "concatenated_vector_map_stats.csv:md5,0649358705d8af6e9a61dc8edeeb5b58"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-15T12:25:52.360597"
    }
}